benchmarks/distributed/rpc/parameter_server/trainers/RpcTrainerBase.py

```diff
@@ -0,0 +1,54 @@
+import torch
+import torch.distributed.rpc as rpc
+
+from .TrainerBase import TrainerBase
+
+
+class RpcTrainerBase(TrainerBase):
+
+    RPC_FUT_METRIC = "rpc_fut_metric"
+    RPC_FUT = "rpc_fut"
+    RPC_METRIC = "rpc_metric"
+    RPC = "rpc"
+
+    def __init__(self, rank):
+        super().__init__(rank)
+
+    def send_async_request(self, key, rref, server_method, *args, cuda=True):
+        self.record_rpc_fut_start(key, f"{self.RPC_FUT}_{server_method.__name__}", cuda)
+        fut = rpc.rpc_async(
+            rref.owner(),
+            server_method,
+            args=args
+        )
+
+        def request_time_callback(fut):
+            self.record_rpc_fut_end(key)
+            return fut.wait()
+
+        return fut.then(request_time_callback)
+
+    def send_sync_request(self, key, rref, server_method, *args, cuda=True):
```

Comment:
this API is not used in this PR. Let's only introduce new API when they are needed.

Reply:
okay will remove.